### ISSUE TITLE
Implement a script to mirror test GCP project

### ISSUE THREAD

**Issue by mshudrak** [google/gcp_scanner#37]
Currently, there is no simple way for developers to mirror GCP project used in testing. However, this might be required for developers in order to locally test their version of GCP scanner or even debug tests itself. 

It would be great to have a script that can deploy a mirror/copy of our test GCP project for new developers. We also need to provide documentation to ensure easy setup.

**Comment by sudiptob2** [comment]
I am interested in this topic as we discussed already in the google group.

**Comment by sudiptob2** [comment]
I think we need following DOD to cover

- [ ] Create a script to deploy the project in the GCP
- [ ] Documentation on how to use the script
- [ ] Documentation on how to run the tests against the project

One of my concerns is, to setup up the project `enabling billing` might be required in the GCP. Not everyone will be interested to add billing information.
Do you have any workaround in mind for this

**Comment by mshudrak** [comment]
Yea, that's a decent plan. As for billing, I don't think I have a simple solution for that. GCP offers free $300 credit https://cloud.google.com/free/docs/free-cloud-features#free-trial for new customers but that's obviously does not solve the problem. 

Alternatively, we might potentially think about safe way to grant any new developer access to main test GCP project (e.g. viewer access).

**Comment by ArkaprabhaChakraborty** [comment]
>Alternatively, we might potentially think about safe way to grant any new developer access to main test GCP project (e.g. viewer access).

This sounds better

**Comment by sudiptob2** [comment]
yes, 300$ credit is there but it still needs a credit card to avail that. 
Later solution seems feasible

**Comment by ArkaprabhaChakraborty** [comment]
Actually, we can kinda have an "OAuth2"-ed gateway kinda logic for contributors and that "token" can be used to run the tests on an "instance" and return the results probably?

**Comment by ArkaprabhaChakraborty** [comment]
It's a bad idea that came in my mind :)

**Comment by sudiptob2** [comment]
I think I can do a POC regarding a `safe way to grant any new developer access`

**Comment by ArkaprabhaChakraborty** [comment]
> I think I can do a POC regarding a `safe way to grant any new developer access`

On a custom project? That POC can be done tho.

**Comment by sudiptob2** [comment]
> On a custom project? That POC can be done tho.

Yes, basically we need to check if we can safely grant user read access to `the project` that GitHub action uses to run the tests.

**Comment by sudiptob2** [comment]
> On a custom project? That POC can be done tho.

Yes, basically we need to check if we can safely grant user read access to `the project` that GitHub action uses to run the tests.

**Comment by mshudrak** [comment]
Let me think about this one. It is not super clear on what would be the best option here.

**Comment by mhdsysy** [comment]
TBH, a well-documented step by step guide for setting up GCP for local testing is more than enough if allowing certain group of users to branch out from this project directly and trigger the actions from their branches is too much power.

**Comment by mshudrak** [comment]
@mhdsysy Yes but the issue is that GCP requires credit card in order to setup a project which might be a problem for some folks.
